lib/internal/test_runner/test.js

```diff
@@ -760,8 +760,10 @@ class Suite extends Test {
 
     try {
       const { ctx, args } = this.getRunArgs();
+      const runArgs = ArrayPrototypeSlice(args);
+      ArrayPrototypeShift(runArgs, ctx);
       this.buildSuite = PromisePrototypeThen(
-        PromiseResolve(this.runInAsyncScope(this.fn, ctx, args)),
+        PromiseResolve(ReflectApply(this.runInAsyncScope, this.fn, runArgs)),
```

Comment:
```suggestion
        PromiseResolve(ReflectApply(this.runInAsyncScope, this, runArgs)),
```

Reply:
Unless there’s a good reason for changing it (in which cas, please add a comment), this code should be using `this` instead of `this.fn` to be equivalent to what there was before.

Reply:
but... this.fn was oassed before?

Reply:
@benjamingr `this.runInAsyncScope` was invoked with a receiver of `this`, and `this.fn, ctx, args` as the argument list. This PR changes it to invoke it with Reflect.apply, which needs to use `this` as the receiver (not `this.fn`), and then `runArgs` should be `[this.fn, ctx, ...args]`.